### PR TITLE
Exception handler cleanup:

### PR DESCRIPTION
**Pull request by nadberezny**
1. Remove superfluous EspExceptionHandler abstractions.
2. Remove deprecated FlinkEspExceptionHandler implementations.

### CODE REVIEW

File: engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/api/exception/FlinkEspExceptionHandler.scala [hunk 1]
```diff
@@ -13,13 +16,22 @@ object FlinkEspExceptionHandler {
   }
 }
 
-trait FlinkEspExceptionHandler extends EspExceptionHandler {
+trait FlinkEspExceptionHandler extends Lifecycle{
```
Comment (lciolecki) [inline engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/api/exception/FlinkEspExceptionHandler.scala:19]: FlinkNuExceptionHandler

File: api/src/main/scala/pl/touk/nussknacker/engine/api/exception/EspExceptionInfo.scala [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package pl.touk.nussknacker.engine.api.exception
+
+import pl.touk.nussknacker.engine.api.Context
+
+case class EspExceptionInfo[T <: Throwable](nodeId: Option[String], throwable: T, context: Context) extends Serializable
```
Comment (lciolecki) [inline api/src/main/scala/pl/touk/nussknacker/engine/api/exception/EspExceptionInfo.scala:5]: NuExceptionInfo?

File: docs/Changelog.md [hunk 3]
```diff
@@ -61,6 +61,8 @@ and `GenericNodeTransformation.implementation` is not invoked in this case
 * [#2474](https://github.com/TouK/nussknacker/pull/2474) Interpreter return type changed from `F[Either[List[InterpretationResult], EspExceptionInfo[_ <: Throwable]]]` to `F[List[Either[InterpretationResult, EspExceptionInfo[_ <: Throwable]]]]`.
   Hence, e.g. multiple branches in Graph can be evaluated, both positively and negatively at the same time.
 * [#2441](https://github.com/TouK/nussknacker/pull/2441) avro sink supports defaults of primitive avro types
+* [#2498](https://github.com/TouK/nussknacker/pull/2498), [#2499](https://github.com/TouK/nussknacker/pull/2499), [#2503](https://github.com/TouK/nussknacker/pull/2503), [#2539](https://github.com/TouK/nussknacker/pull/2539) EspExceptionHandler is removed from ProcessConfigCreator.
```
Comment (mproch) [inline docs/Changelog.md:64]: Let's move it to 1.2.0 :) also - probably former changes?

File: engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/api/exception/FlinkEspExceptionHandler.scala (hunk 1 above)
Comment (mproch) [reply engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/api/exception/FlinkEspExceptionHandler.scala:19]: Do we need this trait at all in API? I think the idea is to have only `ConfigurableExceptionHandler` and `ExceptionConsumer` as part of the API?

File: docs/Changelog.md (hunk 3 above)
Comment (nadberezny) [reply docs/Changelog.md:64]: :+1:

File: engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/api/exception/FlinkEspExceptionHandler.scala (hunk 1 above)
Comment (nadberezny) [reply engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/api/exception/FlinkEspExceptionHandler.scala:19]: In fact we don't need it. I'll remove it :+1:

File: api/src/main/scala/pl/touk/nussknacker/engine/api/exception/EspExceptionInfo.scala (hunk 2 above)
Comment (nadberezny) [reply api/src/main/scala/pl/touk/nussknacker/engine/api/exception/EspExceptionInfo.scala:5]: :+1:

File: engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/api/exception/FlinkEspExceptionHandler.scala (hunk 1 above)
Comment (nadberezny) [reply engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/api/exception/FlinkEspExceptionHandler.scala]: removal: https://github.com/TouK/nussknacker/pull/2539/commits/33b2878e90a3119f3b5f01f90a9e88517442137f
rename: https://github.com/TouK/nussknacker/pull/2539/commits/8edb4673a182a5e75b12c72386fbb128f4977868

File: api/src/main/scala/pl/touk/nussknacker/engine/api/exception/EspExceptionInfo.scala (hunk 2 above)
Comment (nadberezny) [reply api/src/main/scala/pl/touk/nussknacker/engine/api/exception/EspExceptionInfo.scala]: done: https://github.com/TouK/nussknacker/pull/2539/commits/dde2146c0dbd154458362886b977036d7ef3c789

File: docs/Changelog.md (hunk 3 above)
Comment (nadberezny) [reply docs/Changelog.md]: done: https://github.com/TouK/nussknacker/pull/2539/commits/40365223a88fbfb5bdb2525f485cd3fc79922069

**Review by mproch — APPROVED** [review]
:+1:  thanks :)

**Pull request merged**